.gitignore

```diff
@@ -1,20 +1,52 @@
-###> Docker ###
-docker-compose.override.yml
-docker-compose.override.yaml
-###< Docker ###
+# Comment out the following line, if you wish to store your SQLite database in Git
+/var/data/*
 
 ###> bolt/core ###
 config/bolt/*_local.yaml
 /config/packages/local
 /config/routes/local
 /public/files/*
 !/public/files/.gitkeep
+!/public/files/index.html
 /public/themes/*
 !/public/themes/.gitkeep
 /public/thumbs/*
 !/public/thumbs/.gitkeep
+!/public/thumbs/index.html
 ###< bolt/core ###
 
+### NPM and frontend assets building cruft
+node_modules/
+npm-debug.log
+yarn-error.log
+build_assets.sh
+
+### File-system cruft and temporary files
+__*
+._*
+.*.swp
+.swp
+*.lock
+!composer.lock
+!symfony.lock
+*lock.json
+!/package-lock.json
+.buildpath
+
+### Platfom-specific files
```

Comment:
```
### Platfom-specific files
.DS_Store
thumbs.db
Vagrantfile
.vagrant*
.idea
.vscode/*
appveyor.yml
```

This is bad practice, one should never have to ignore the files or folders in the development platform. The project does not know what the developer is using as IDE, as Operating system .....
**It is not up to the project to carry this information.**

On the other hand, having clearer documentation to increase users' skills in GIT good practice is not negligible.

The good GIT practice is to have a gitignore at the level of the overall configuration of your machine.

For example:
`$ nano ~/.gitconfig`
```
[user]
	name = Lastname Firstname
	email = my-email@domain.com
[color]
	ui = true
[core]
	editor = vi
	excludesfile = ~/.gitignore_global
```


`$ nano ~/.gitignore_global`
```
###################
# Compiled source #
###################
*.com
*.class
*.dll
*.exe
*.o
*.so

############
# Packages #
############
# it's better to unpack these files and commit the raw source
# git has its own built in compression methods
*.7z
*.dmg
*.gz
*.iso
*.jar
*.rar
*.tar
*.zip

######################
# Logs and databases #
######################
*.log
*.sql
*.sqlite

######################
# OS generated files #
######################
.DS_Store
.DS_Store?
._*
.Spotlight-V100
.Trashes
ehthumbs.db
Thumbs.db
*~
.*.swp

#######
# IDE #
#######
# for NetBeans
nbproject
# for *Storm
.idea
# for VScode
.vscode
.project
.buildpath
.setting/org.eclipse.php.core.prefs
.settings/org.eclipse.wst.common.project.facet.core.xml
```

Reply:
Do you suggest removing the platform-specific files and folders?

While I agree that it's not up to this project to guesstimate what people use, I think the more we can clean up _for_ people in advance, the better.

In my experience, people who are not that good with git/bolt will commit _all_ the changes to get their project working, and they'll be confused about IDE files and such. If anyone wants to commit their `.idea` file, I'd guess they are experienced enough to know what they are doing, and will know to remove it from `.gitignore`.

Reply:
This has been a discussion back and forth for many years. I'm with Ivo on this one.

Yes, it's better practice if the _user_ has a global `.gitignore`, but because a lot of users don't have one, we'll add them in our project-specific `.gitignore` :-)

Reply:
ok